client/client.go

```diff
@@ -217,15 +217,28 @@ func (ar *ApiRequest) Do(req *http.Request) (*http.Response, error) {
 		req.URL.Host = host
 		req.Host = host
 
+		log.Debugf("Connecting to server %s", host)
+
+		var body io.ReadCloser
+		if req.GetBody != nil {
```

Comment:
Question:
It's late, and I'm leaving now, and was not fully able to grasp this. But could another workaround have been buffering this outside of the loop in a bytes buffer, and resetting it on every invocation?

Reply:
Possibly, but we would be "swimming against the current", so to speak. The Golang http library states that it may also call `GetBody` (for example when following redirects), so it has to be kept working. So we might as well call it as well, to be consistent, and to test it.